README.md

```diff
@@ -1,82 +1,294 @@
+
+## Step 2: Build a circuit statement
+
+Proof Market works with circuits in the form of circuit statements.
+A statement is basically a JSON with the circuit and various metadata that identifies it.
+
+The `build_statement` command will build a circuit statement from the circuit that we compiled earlier:
+
+```bash
+./scripts/run.sh --docker build_statement
 ```
 
-# Zero-knowledge proof workflow
+The `build_statement` command does the following:
 
-# Step 1 : Build the intermediate representation (IR) file
-``` 
-mkdir build && cd build
-cmake .. && make template
+1. Runs a new container based on `nilfoundation/proof-market-toolchain`.
+2. In the container, runs `prepare_statement.py` to produce a circuit statement.
+
+<details>
+  <summary><code>[manual mode]</code></summary>
+
+To build a statement, we will use the `prepare_statement.py` script,
+which is a part of the [Proof Market toolchain](https://github.com/nilfoundation/proof-market-toolchain).
+
+First, start a new container with the Proof Market toolchain.
+(Remember to exit the `zkllvm` container with `exit` command or start a new console session):
+
+```bash
+docker run -it --rm \
+    --platform=linux/amd64 \
+    --volume $(pwd):/opt/zkllvm-template \
+    --volume $(pwd)/.config:/.config/ \
+    --volume $(pwd)/.config:/root/.config/ \
+    --user $(id -u ${USER}):$(id -g ${USER}) \
+    ghcr.io/nilfoundation/proof-market-toolchain:0.0.31
+```
+
+> The `.config` directory is where you will put the credentials to the Proof Market later on.
+Two extra volume mounts make this directory available in places where
+parts of the Proof Market toolchain might look for it.
+
+Now pack the circuit into a statement:
+
+```bash
+cd /opt/zkllvm-template/
+python3 \
+    /proof-market-toolchain/scripts/prepare_statement.py \
+    -c /opt/zkllvm-template/build/src/template.ll \
+    -o /opt/zkllvm-template/build/template.json \
+    -n template \
+    -t placeholder-zkllvm
+  
+# -c: path to the circuit file
+# -o: path to write the statement file
+# -n: statement name
+# -t: type of proofs that will be generated with this statement
+# (Placeholder is the name of our proof system, see
+# https://crypto3.nil.foundation/papers/placeholder.pdf)
+```
+</details>
+
+As a result, we have the circuit statement file `./build/template.json`.
+Later we will use it to generate a proof locally, and then push the statement to the Proof Market.
+
+## Step 3: Produce and verify a proof locally
+
+Now we have everything ready to produce our first proof.
+We'll use the `proof-generator` CLI, which is a part of the Proof Market toolchain.
+
+```bash
+./scripts/run.sh --docker prove
 ```
-You should have a circuit IR file called `template.ll` 
 
-# Step 2: Setup proof market user/toolchain
-Please navigate out of the `zkllvm-template` repository
+<details>
+  <summary><code>[manual mode]</code></summary>
+
+Continue in the `proof-market-toolchain` container that you made in step 2.
```

Comment:
```suggestion
Continue in the `proof-market-toolchain` container that you made in Step 2.
```

Reply:
It's a capital letter just because it's the first word in a sentence.